### PR TITLE
Fixed symmetric serialization of JsValue not equaling itself.

### PR DESCRIPTION
**Pull request by jeffmay**
@patrick-bohan @rwhiten1 @slustbader

This fixes the generators to prevent the generation of JsValue's that can not be serialized and deserialized to equal themselves.

### CODE REVIEW

File: playJsonTests/src/main/scala/play/api/libs/json/scalacheck/JsValueGenerators.scala [hunk 1]
```diff
@@ -49,18 +53,51 @@ trait JsValueGenerators {
   }
 
   /**
+   * The Jaxson parser has trouble with very large exponents of BigDecimals.
+   *
+   * I couldn't quite pin the problem down to precision, scale, or both, so
+   */
+  def genSafeBigDecimal: Gen[BigDecimal] = {
```
Comment (arkban) [inline playJsonTests/src/main/scala/play/api/libs/json/scalacheck/JsValueGenerators.scala:60]: Add ()

**Comment by arkban** [comment]
LGTM

**Comment by slustbader** [comment]
LGTM

File: build.sbt [hunk 2]
```diff
@@ -56,7 +56,7 @@ lazy val playJsonOps = project in file("playJsonOps") settings(common: _*) setti
   )
 
 ) dependsOn (
-  playJsonTests % "compile->test"
```
Comment (ghost) [inline build.sbt:59]: :+1:

**Pull request merged**